packages/components/src/keyboard-shortcuts/index.js

```diff
@@ -21,7 +21,22 @@ class KeyboardShortcuts extends Component {
 		const { keyTarget = document } = this;
 
 		this.mousetrap = new Mousetrap( keyTarget );
+
 		forEach( this.props.shortcuts, ( callback, key ) => {
+			if ( process.env.NODE_ENV === 'development' ) {
+				const keys = key.split( '+' );
+				const modifiers = new Set( keys.filter( ( value ) => value.length > 1 ) );
+				const hasAlt = modifiers.has( 'alt' );
+				const hasShift = modifiers.has( 'shift' );
+
+				if (
+					( modifiers.size === 1 && hasAlt ) ||
+					( modifiers.size === 2 && hasAlt && hasShift )
+				) {
+					throw new Error( `Cannot bind ${ key }. Alt and Shift+Alt modifiers are reserved for character input.` );
```

Comment:
I guess it's _expected_ this should crash the application (to make the problem exceedingly obvious)? As there's been some prior conversation at #13405 and alternative approaches at #14151 to address these sorts of "wrong usage" violations, it seems we might consider having a unified approach to when and how they should be introduced. Personally I'd favor ESLint rules since they can be captured at build-time than at runtime, but I could grant that (a) there's more effort involved in creating the rules ([though not always](https://github.com/WordPress/gutenberg/blob/774713c0467779a9651983ec96609a6637811132/.eslintrc.js#L27-L100)) and (b) this assumes that the developer has opted into using ESLint and the custom rules, which is not a given.